core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -153,8 +153,15 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(classOf[SerializableJobConf], new KryoJavaSerializer())
     kryo.register(classOf[PythonBroadcast], new KryoJavaSerializer())
 
-    kryo.register(classOf[GenericRecord], new GenericAvroSerializer(avroSchemas))
-    kryo.register(classOf[GenericData.Record], new GenericAvroSerializer(avroSchemas))
+    kryo.register(classOf[GenericRecord], new GenericAvroSerializer[GenericRecord](avroSchemas))
```

Comment:
Can we get a little more DRY on these lines like:
```
    def registerAvro[T]: Unit = kryo.register(classOf[T], new GenericAvroSerializer[T](avroSchemas))
    registerAvro[GenericRecord]
    registerAvro[GenericData.Record]
    // ...
```

Reply:
Done. Thanks! This looks so much better now!